ros/api/v1/recommendation_ratings.py

```diff
@@ -28,20 +28,28 @@ def post(self):
 
         data = json.loads(request.data)
         inventory_id = data['inventory_id']
-        rating = int(data['rating'])
-        if rating not in [c.value for c in RatingChoicesEnum]:
-            abort(
-                422,
-                message=(
-                    "{} is not a valid value for rating".format(rating)))
+        allowed_choices = [c.value for c in RatingChoicesEnum]
+
+        def invalid_value_msg(val):
```

Comment:
Looks like this is a good contender for decorator validation. 

We can abstract all this code into a nice little function and keep our code pretty and modular.

Just like how we use ` @marshal_with(x_fields)`

Example:

```
def decorator(func):
    def inner():
        print("run something before actual function")
        func()
        print("run something after actual function")
    return inner


@decorator
def func():
    print("something...")


func()
"""
run something before actual function
something...
run something after actual function
"""
```

References:  

- https://levelup.gitconnected.com/validations-in-python-using-metaprogramming-and-decorators-advanced-python-ee4d4278a6b3

- https://stackoverflow.com/questions/50711796/how-to-validate-arguments-using-decorators-in-python

Reply:
Good idea. Added changes, please request to review.